packages/dotnet/src/lib/core/dotnet.client.ts

```diff
@@ -370,6 +384,31 @@ export class DotNetClient {
     return res.stdout.toString();
   }
 
+  async spawnAsyncAndGetOutput(params: string[]): Promise<string> {
+    params = params.map((param) =>
+      param.replace(/\$(\w+)/, (_, varName) => process.env[varName] ?? ''),
+    );
+
+    const { stdout } = await promisify(execFile)(
+      this.cliCommand.command,
+      params,
+      {
+        cwd: this.cwd ?? process.cwd(),
+      },
+    ).catch((e) => {
+      if ('code' in e && 'stderr' in e) {
+        throw new Error(
+          `dotnet execution returned status code ${e.code} \n ${e.message}`,
+        );
+      }
+      throw e;
+    });
+    if (stdout.includes('There are no Project to Project')) {
+      return '';
+    }
+    return stdout;
+  }
```

Comment:
Using `execFile` instead of `spawn` made the code slightly more readable and hopefully, easier to maintain.